cloud-sql-proxy/cloud-sql-proxy.sh

```diff
@@ -233,39 +239,46 @@ readonly KMS_KEY_URI
 DB_ADMIN_PASSWORD_URI="$(/usr/share/google/get_metadata_value attributes/db-admin-password-uri || echo '')"
 readonly DB_ADMIN_PASSWORD_URI
 
-DB_ADMIN_PASSWORD=''
-if [[ -n "${DB_ADMIN_PASSWORD_URI}" ]]; then
+if [[ -n "${DB_ADMIN_SECRET}" ]] ; then
+  gcloud secrets versions access "${DB_ADMIN_SECRET#*:}" \
+      --project="${METASTORE_INSTANCE%%:*}" \
+      --secret="${DB_ADMIN_SECRET%:*}" > /dev/shm/db-pw
+elif [[ -n "${DB_ADMIN_PASSWORD_URI}" ]]; then
   # Decrypt password
-  DB_ADMIN_PASSWORD="$(gsutil cat "${DB_ADMIN_PASSWORD_URI}" |
+  gsutil cat "${DB_ADMIN_PASSWORD_URI}" |
     gcloud kms decrypt \
       --ciphertext-file - \
       --plaintext-file - \
-      --key "${KMS_KEY_URI}")"
+      --key "${KMS_KEY_URI}" > /dev/shm/db-pw
+else
+  touch /dev/shm/db-pw
 fi
-if [[ "${CLOUDSQL_INSTANCE_TYPE}" == "POSTGRES" && -z "${DB_ADMIN_PASSWORD}" ]]; then
+if [[ "${CLOUDSQL_INSTANCE_TYPE}" == "POSTGRES" && -z "$(perl -pe 'chomp' < /dev/shm/db-pw)" ]]; then
```

Comment:
this test could instead be done with `wc -c` to eliminate the final instance of the variable being stored in bash strings